setup.py

```diff
@@ -105,21 +105,40 @@ def add_optimization_flags(self):
         """
 
         env = os.environ.copy()
-        if not setupext.config.getboolean('libs', 'enable_lto', fallback=True):
-            return env
         if sys.platform == 'win32':
             return env
-
+        enable_lto = setupext.config.getboolean('libs', 'enable_lto',
+                                                fallback=None)
+
+        if 'CFLAGS' in os.environ:
+            if '-fno-lto' in os.environ['CFLAGS']:
+                if enable_lto is True:
+                    raise ValueError('Configuration enable_lto=True, but '
+                                     'CFLAGS contains -fno-lto')
+                enable_lto = False
```

Comment:
is there a reason it's not a function?
```python
def check_lto(compiler_flag, enable_lto):
    if '-fno-lto' in os.environ[compiler_flag]:
        if enable_lto is True:
            raise ValueError(f'Configuration enable_lto=True, but {compiler_flag}
                                     contains -fno-lto')
            enable_lto = False`
     return enable_Ito
```
so that 
```python
if 'CFLAGS' in os.environ: 
    enable_lto = check_lto('CFLAGS', enable_lto)
```
or even with the compiler flag fully backed in if that's the only check for it?

Reply:
Yep, that can be done.